File: tests/drivers/node/v1/basic.js
var neo4j = require('neo4j-driver').v1;
var driver = neo4j.driver("bolt://localhost:7687",
                          neo4j.auth.basic("neo4j", "1234"),
                          { encrypted: 'ENCRYPTION_OFF' });
var session = driver.session();

function die() {
  session.close();
  driver.close();
  process.exit(1);
}

function run_query(query, callback) {
  var run = session.run(query, {});
  run.then(callback).catch(function (error) {
    console.log(error);
    die();
  });
}

run_query("MATCH (n) DETACH DELETE n", function (result) {
  console.log("Database cleared.");
  run_query("CREATE (alice: Person {name: 'Alice', age: 22})", function (result) {
    console.log("Record created.");
    run_query("MATCH (n) RETURN n", function (result) {
      console.log("Record matched.");
      var alice = result.records[0].get("n");
      if(alice.labels[0] != "Person" || alice.properties["name"] != "Alice"){
        console.log("Data doesn't match!");
        die();
      }
      console.log("All ok!");
      driver.close();
    });
  });
});

File: tests/drivers/java/v1/Basic.java
import static org.neo4j.driver.v1.Values.parameters;

import java.util.*;
import org.neo4j.driver.v1.*;
import org.neo4j.driver.v1.types.*;

public class Basic {
  public static void main(String[] args) {
    Config config = Config.build().withoutEncryption().toConfig();
    Driver driver =
        GraphDatabase.driver("bolt://localhost:7687", AuthTokens.basic("neo4j", "1234"), config);

    try (Session session = driver.session()) {
      StatementResult rs1 = session.run("MATCH (n) DETACH DELETE n");
      System.out.println("Database cleared.");

      StatementResult rs2 = session.run("CREATE (alice: Person {name: 'Alice', age: 22})");
      System.out.println("Record created.");

      StatementResult rs3 = session.run("MATCH (n) RETURN n");
      System.out.println("Record matched.");

      List<org.neo4j.driver.v1.Record> records = rs3.list();
      org.neo4j.driver.v1.Record record = records.get(0);
      Node node = record.get("n").asNode();
      if (!node.get("name").asString().equals("Alice") || node.get("age").asInt() != 22) {
        System.out.println("Data doesn't match!");
        System.exit(1);
      }

      System.out.println("All ok!");
    } catch (Exception e) {
      System.out.println(e);
      System.exit(1);
    }

    driver.close();
  }
}
